floodviz/map_utils.py

```diff
@@ -36,11 +65,29 @@ def site_dict(site_list, url_prefix):
 
 def write_geojson(filename, data):
     """Writes site data to a .json file so it can be mapped
-
        Args:
-           filename: the file to be written to
-           data: the data to be written to the file
+           filename: A string naming the file to be written to
+           data: A list of dicts with data to be written to the file
     """
+    if not type(filename) == str:
+        print("Error: filename must be string")
+        return;
+
+    if not filename.endswith(".json"):
+        print("Error: outfile must be .json")
+        return
+
+    if not data:
```

Comment:
Similarly, the user of the function know data is a list. If the list is empty then write an empty array to the json file. FYI the first if statement would be true if data is None so the second if statement would never be executed. All three of these if statements are not needed.